src/apigateway/commands/copyUrl.ts

```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Window } from '../../shared/vscode/window'
+import { Env } from '../../shared/vscode/env'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { RestApiNode } from '../explorer/apiNodes'
+import * as picker from '../../shared/ui/picker'
+import * as vscode from 'vscode'
+import { ext } from '../../shared/extensionGlobals'
+import { Stage } from 'aws-sdk/clients/apigateway'
+import { ApiGatewayClient } from '../../shared/clients/apiGatewayClient'
+import { RegionProvider } from '../../shared/regions/regionProvider'
+import { DEFAULT_DNS_SUFFIX } from '../../shared/regions/regionUtilities'
+
+const COPY_URL_DISPLAY_TIMEOUT_MS = 2000
+
+interface StageInvokeUrlQuickPick extends vscode.QuickPickItem {
+    // override declaration so this can't be undefined
+    detail: string
+}
+
+export async function copyUrlCommand(
+    node: RestApiNode,
+    regionProvider: RegionProvider,
+    window = Window.vscode(),
+    env = Env.vscode()
+): Promise<void> {
+    const region = node.regionCode
+    const dnsSuffix = regionProvider.getDnsSuffixForRegion(region) || DEFAULT_DNS_SUFFIX
+    const client: ApiGatewayClient = ext.toolkitClientBuilder.createApiGatewayClient(region)
+
+    const stages: Stage[] = (await client.getStages(node.id)).item || []
+    const quickPickItems = stages.map<StageInvokeUrlQuickPick>(stage => ({
+        label: stage.stageName!!,
+        detail: buildDefaultApiInvokeUrl(node.id, region, dnsSuffix, stage.stageName!!),
+    }))
+
+    const quickPick = picker.createQuickPick({
```

Comment:
1. if `quickPickItems` is empty, short circuit the user to a message instead of showing an empty picker
1. if `quickPickItems` contains one item, would the flow make sense to bypass the picker?

Reply:
1. sure
2. that's what i was considering, but was thinking it'd be jarring if the API call took a while and the only indicator of success was a tiny statusbar text that they'd may not necessarily be looking for. any ideas?

Reply:
If the API call generally takes a while (account for things like slow wifi and typical backend response time), posting to the status bar, and possibly placing a busy indicator on the "aws panel" icon for the duration of that process sounds reasonable.